dask/array/blockwise.py

```diff
@@ -160,30 +160,30 @@ def blockwise(
         if not isinstance(v, tuple):
             v = (v,)
         chunkss[k] = v
-    arginds = list(zip(arrays, args[1::2]))
 
-    for arg, ind in arginds:
-        if hasattr(arg, "ndim") and hasattr(ind, "__len__") and arg.ndim != len(ind):
-            raise ValueError(
-                "Index string %s does not match array dimension %d" % (ind, arg.ndim)
-            )
-
-    numblocks = {a.name: a.numblocks for a, ind in arginds if ind is not None}
+    arginds = zip(arrays, args[1::2])
```

Comment:
There seems to be places where materialization to `list` is not necessary. Hence removed all such instances.